src/compas/files/base_reader.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+    def __init__(self, location):
+        self.is_url = False
+        self.location = self.check_location(location)
+
+    def read(self, mode='text'):
+        """Returns iterable reading file line by line when given a location,
+           either as pathlib object, string containing path or string containing url
+        Arguments
```

Comment:
since we shoot for a docstring style that is as readable as possible in code as well i would advocate for a newline here

Reply:
i personally also prefer "Parameters" over "Arguments"

Reply:
Technically, the summary should be one-line only. Then you can have a new-line and expand on it with further detail.

Reply:
Ok, good points, fixed.

https://github.com/compas-dev/compas/blob/d27f338942d4d5109ad888c80534bd1ca8398193/src/compas/files/base_reader.py#L61